node/setup.go

```diff
@@ -251,13 +253,38 @@ func createMempoolAndMempoolReactor(
 	// allow empty string for backward compatibility
 	case cfg.MempoolTypeFlood, "":
 		logger = logger.With("module", "mempool")
+		options := []mempl.CListMempoolOption{
+			mempl.WithMetrics(memplMetrics),
+			mempl.WithPreCheck(sm.TxPreCheck(state)),
+			mempl.WithPostCheck(sm.TxPostCheck(state)),
+		}
+		if config.Mempool.PublishEventPendingTx {
```

Comment:
All this logic is intended to bash transactions and produce events from batches of transactions?

If you check how the `PublishEventTx` is used: 

https://github.com/cometbft/cometbft/blob/2dd5209177270fde125c71aeedda9819046f92f0/state/execution.go#L749-L758

you will see that no batching is performed, as this is the role of the event bus by itself. It expect to take multiple events, put them in a queue and make sure that the queue is consumed fast enough, otherwise events are dropped.

Reply:
It is also used to prevent potential blocking of event publish, which can affect the performance of mempool.

Reply:
Why?

`PublishWithEvents` adds the event to the `s.cmds` channel of the pubsub server. This server contains an infinite `loop` that for publish events calls a `send` method with:

https://github.com/cometbft/cometbft/blob/2dd5209177270fde125c71aeedda9819046f92f0/libs/pubsub/pubsub.go#L420-L432

So, essentially, if the subscription has a buffer, there is no blocking here. Otherwise, any slow client subscribing to the `Tx` event would crash the node.

Reply:
Again, when a block is committed, all txs in the block are published using the `Tx` event. This is more publishing calls than any solution inside the mempool. And this does not break, provided it is properly used.

If you say that this method is not efficient and can be improved, I will very likely agree. But this PR is not about improving the pubsub service, is about creating a new event to be published using the existing service.

Reply:
That's the [origin](https://github.com/cometbft/cometbft/blob/fba9b434cdd8b0a355291b2d10086fe3e6c5037b/node/setup.go#L263) way but there's a [concern](https://github.com/cometbft/cometbft/pull/3111#discussion_r1615819167) about blocking behaviour.

Reply:
Yes, I saw @sergio-mena comment. But I still wouldn't solve the problem in this way, namely, by adding a sub-routine inside the `node` package to batch notifications about txs added to the mempool.

Reply:
Any suggestion is welcome.

Reply:
I gave my suggestion: let `pubsub` handle this and set a decent buffer length for this subscription. Then warns users that with this option they can observe performance issues. We attempt to measure the impact in performance using some test/benchmark units.

@sergio-mena does not agree with me, so I will probably discuss this with him synchronously.